library/common/engine.cc

```diff
@@ -93,6 +93,19 @@ Engine::~Engine() {
   main_thread_.join();
 }
 
+void Engine::recordCounter(std::string elements, uint64_t count) {
+  if (server_) {
+    server_->dispatcher().post([this, elements, count]() -> void {
+      static const std::string client = "client";
```

Comment:
No non-pod statics (static init fiasco). Moreover what you really want here since this string is known at compile-time is to save a StatName for "client" using either a StatNamePool or StatNameManagedStorage. You want to do that in a context object that is created once at process startup and then re-used. Then you have no thread contention issues.